cli/src/cast.rs

```diff
@@ -500,6 +500,59 @@ async fn main() -> eyre::Result<()> {
             let selector = contract.abi().functions().last().unwrap().short_signature();
             println!("0x{}", hex::encode(selector));
         }
+        Subcommands::FindBlock { timestamp, rpc_url } => {
+            let ts_target = U256::from(timestamp);
+            let provider = Provider::try_from(rpc_url)?;
+            let last_block_num = provider.get_block_number().await?;
+            let cast_provider = Cast::new(provider);
+            let ts_block = cast_provider.timestamp(last_block_num).await?;
+            let ts_block1 = cast_provider.timestamp(1).await?;
+            let block_num = if ts_block.lt(&ts_target) {
+                // If the most recent block's timestamp is below the target, return it
+                last_block_num
+            } else if ts_block1.gt(&ts_target) {
+                // If the target timestamp is below block 1's timestamp, return that
+                U64::from(1)
+            } else {
```

Comment:
hm, that doesn't seem directly applicable here though, since the three cases come from comparing x to two different variables, not just 1:
- Case 1 : `x < timestamp_block_1`
- Case 2: `x > timestamp_block_latest`
- Case 3: `timestamp_block_1 <= x <= timestamp_block_latest`